### ISSUE TITLE
Ability to manually define "Primary identifier" by ADD operartion.

### ISSUE THREAD

**Issue by Me-okane** [ExLibrisGroup/alma-csv-user-load#102]
We are currently using the "Primary identifier" for our SAML login.   Manually added users are able to use  uid@domain.de as a Primary Identifier.   The CSV import with  "ADD" operation creates its own (numerical) ID.

Is there a way to make the CSV import script accept a given  Primary Identifier ?

**Comment by mgobat** [comment]
Hi @Me-okane. You should be able to create an "Add" profile specifying the primary identifier in the field mapping. For example, here's a screenshot of my test profile:
![addUsers](https://github.com/user-attachments/assets/ccdd152b-0efc-4c75-908c-6d0ea2af3293)
Note that the third entry indicates that the "Primary Identifier" (Header) in my csv input file should map to the "Primary ID" (Name). 

Then here is the header in my csv file:
`First Name;Last Name;Primary Identifier;User Group;Preferred Language;Expiration Date;Password;Address;Address Type;Email;Email Type;Job Description`
And here is the user record in the csv file:
`Mark;Twain;mtwain@clarivate.com;ST;en;2099-12-31;TestAlma04;My address here;work;mtwain@gmail.com;work;VP`

The intent here is for mtwain@clarivate.com to be the primary identifier for the "Mark Twain" user. 

Then after running the Cloud App using this csv file, I get a message that the user was created. The message reflects the newly created user's primary id:
![userAdded](https://github.com/user-attachments/assets/f7ce3349-98ae-41af-bccf-d4f5539c7110)
Then when I search for the user by primary id "mtwain@clarivate.com", I get the newly added user:
![userSnippet](https://github.com/user-attachments/assets/7e707d4d-b60d-4751-ab63-0ecf9262a59c)

Does this help, or is there something different about your use case that I'm not considering?  Thank you.

**Comment by Me-okane** [comment]
Thank you for the help.   Here is what I have tried:
with:

givenname;sn;uniqueId;mail;employeetype;Record Type;dateOfRevocation
Maxine;Musterfrau;418832540@example.de;maxine.musterfrau@example.de;Studierende;PUBLIC;2025-10-31

I get Success but the User is created with a System generated Primary ID and not what I set in the CSV.
Could it have to do with being and "EXTERNAL"  User?  


![Alma_import-definitions](https://github.com/user-attachments/assets/f93be204-40d0-4780-9b8c-3e10ebc4921b)

![Alma_import_result-PROCESS](https://github.com/user-attachments/assets/d4a6ad7a-679f-47b0-96a8-3edced124494)

![Alma_import_result-USER](https://github.com/user-attachments/assets/36b5af47-64b3-451d-8647-0aad23b6da43)
